### PR TITLE
Webpack buildfiles for Joomla extensions

### PR DESCRIPTION
**Pull request by anibalsanchez**
Hi,
This PR is just a note to link this repository with Webpack build files for Joomla extensions.

The builder automatically generates a full package with every possible extension inside.

CC: @yvesh 

Regards,
Anibal

### CODE REVIEW

**Comment by mbabker** [comment]
This PR is not in a state where it can be considered for acceptance.

- Copyright is "owned" by third party
- GPL v3 license
- What this produces does not follow the project's code style standards

If you want to release this on your own accord then go for it, but as is this is not in a state to be accepted to and maintained by the Joomla project.

**Comment by anibalsanchez** [comment]
Hi,

I can modify the points to comply with the project general guidelines... if it is in the interest of the project and it will be accepted.

If you prefer to keep the boilerplate in the current form, feel free to close the PR.

**Comment by mbabker** [comment]
I don't actually use the boilerplate stuff, so I have no preference either way.  I just see all the PR notifications and when looking at this one those things jumped out at me right away.

**Comment by anibalsanchez** [comment]
okok, let's wait if there's interest.

**Comment by MastersOfMedia** [comment]
I’d really appreciate if this could be brought to Joomla! standards.

Erik

**Comment by OptikWebCreative** [comment]
Me too :D

**Comment by roland-d** [comment]
Hello @anibalsanchez I never received a notification so I did not see this until recently. What does this do for the boilerplates?

**Comment by anibalsanchez** [comment]
Hi @roland-d ,

My original idea was to extend the original repo and include all the available types of extensions... but after a while, I also included automation to build extensions; in a way that it is easy to delete a folder or some files and just pack them with `npm run build`. 

In a second step, I also added templating, composer support, webpack and JavaScript App support to complete the picture. There are also [handy Bash scripts](https://github.com/anibalsanchez/extly-boilerplate-buildfiles-for-joomla/blob/master/README.md#handy-bash-scripts) to replace *Foo* strings and clean the packed libraries. 

Of course, the project has become opinionated. For instance, there isn't an extension type **app** in Joomla, but the project has a first-level folder to support a JavaScript App for Joomla. It would be great to have an official extension type, meanwhile, we can distribute them in the library media folder.

At this point, I'm using the project myself and I'm porting my extensions to this new generic package organization. The objective is still having a complete method to build an extension, being flexible, without becoming too complex.

On the long run, the next logical step would be the implementation of a generic build server, to simplify how a repo can be connected with a extensions build server.

To sum up, the results of the process is a super-flexible extension buildfiles for Joomla, but I think that the PR is now beyond of the scope of your repo of starter files.

**Comment by roland-d** [comment]
Hello @anibalsanchez 

This idea is great but I do believe it is out of scope for the boilerplates as it is not a boilerplate. However as a separate repo I think it will be a great addition for our community. If we can have a separate repo with this in it I am sure users will find it and use it.

Now, I will close this PR. Thank you for your interest.
